assets/javascripts/discourse/components/chat-toggle-membership-btn.js

```diff
@@ -0,0 +1,91 @@
+import Component from "@ember/component";
+import I18n from "I18n";
+import { inject as service } from "@ember/service";
+import { popupAjaxError } from "discourse/lib/ajax-error";
+import ChatApi from "discourse/plugins/discourse-chat/discourse/lib/chat-api";
+import { action, computed } from "@ember/object";
+
+export default class ChatToggleMembershipBtn extends Component {
+  @service chat;
+  tagName = "";
+
+  channel = null;
+  afterToggle = null;
+  isLoading = false;
+  labelType = null;
```

Comment:
I am not sure if it is good to have so many options, but we have a lot of different join/leave channel button styles.

Chat browse page:

![image](https://user-images.githubusercontent.com/920448/179447081-9ae42af1-5303-4c66-85c7-ed8f79c4d39a.png)
![image](https://user-images.githubusercontent.com/920448/179447095-5fe0b3b9-f993-4edc-9978-34a5b4e51a9c.png)

Leave channel button for DMs

![image](https://user-images.githubusercontent.com/920448/179447126-139ae1fa-5c97-4a41-b545-179971c94478.png)

Preview join channel button:

![image](https://user-images.githubusercontent.com/920448/179447165-1868c132-c18e-4ade-9b30-04d95a32707f.png)

About channel:

![image](https://user-images.githubusercontent.com/920448/179447185-7f438d59-8d85-4464-b677-43ba25c13708.png)